subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitSpec.java

```diff
@@ -15,24 +15,30 @@
  */
 package org.gradle.api.internal.tasks.testing.junit;
 
+import org.gradle.api.internal.tasks.testing.filter.TestFilterSpec;
+
+import java.io.Serializable;
 import java.util.Set;
 
-public class JUnitSpec extends AbstractJUnitSpec {
+public class JUnitSpec implements Serializable {
```

Comment:
Should this, `JunitPlatformSpec` and `TestNGSpec` share a common `AbstractTestExecutorSpec` base class?  They all involve filtering.  This could then become a method in `TestClassProcessor` - `AbstractTestExecutorSpec getSpec()` to closer unite these 2 parallel heirarchies.

Reply:
Unless we're going to have some common logic which consumes these filters generically, I'm not sure an abstraction like this is really useful.